### ISSUE TITLE
Source Shopify:  value "153276284994" is out of range for type integer

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#2393]
Normalization on data coming from the Shopify source is failing due to the following exception:

```
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 - [33mDatabase Error in model custom_collections (models/generated/airbyte_tables/public/custom_collections.sql)[0m
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 -   value "153276284994" is out of range for type integer
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/public/custom_collections.sql
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 - 
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 - [33mDatabase Error in model products (models/generated/airbyte_tables/public/products.sql)[0m
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 -   value "4292979818562" is out of range for type integer
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/public/products.sql
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 - 
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 - [33mDatabase Error in model metafields (models/generated/airbyte_tables/public/metafields.sql)[0m
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 -   value "11227676934210" is out of range for type integer
2021-03-10 19:56:04 INFO (/tmp/workspace/24/0) LineGobbler(voidCall):69 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/public/metafields.sql

```

Full logs: [logs-24-0.txt](https://github.com/airbytehq/airbyte/files/6118637/logs-24-0.txt)

## Steps to Reproduce
TODO

## Severity of the bug for you
high -- unable to use the source

## Airbyte Version
0.16.1-alpha

## Connector Version (if applicable)
airbyte/destination-postgres:0.1.16
airbyte/source-shopify-singer:0.1.8

## Additional context
Reported via Slack DM by @DominusKelvin

**Comment by sherifnada** [comment]
same as #2349 

I looked into this. The problem happens because our normalization module automatically uses [`int`](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/cross_db_utils/datatypes.sql#L85) (as opposed to [big_int](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/cross_db_utils/datatypes.sql#L71)) for numeric types. It's not possible to know when to use which because JsonSchema does not communicate any information about numeric type precisions (and [seems unlikely to](https://github.com/json-schema-org/json-schema-spec/issues/574)). 

Some ideas for a fix: 
1. fix this in the source (same for #2349 ) by changing type to `numeric` -- I'm actually not sure if this will fix because normalization maps `numeric` to `float` and I'm not sure what DBT maps that to on destinations. It could still be insufficient precision. 
2. Always use `bigint` and `numeric`. I'm not sure if this will incur more space on
3. enrich the JSON schema we use with a custom field to indicate precision (although this might fail validators at some point)

**Comment by sherifnada** [comment]
Should be fixed by #2410
